### PR TITLE
Issue 2023

### PR DESCRIPTION
**Pull request by koheiw**
Address #2023

### CODE REVIEW

File: R/dfm_sample.R [hunk 1]
```diff
@@ -5,11 +5,9 @@
 #' @param size a positive number, the number of documents or features to select.
 #'   The default is the number of documents or the number of features, for
 #'   `margin = "documents"` and `margin = "features"` respectively.
-#' @param replace logical; should sampling be with replacement?
-#' @param prob a vector of probability weights for obtaining the elements of the
-#'   vector being sampled.
 #' @param margin dimension (of a [dfm]) to sample: can be `documents` or
 #'   `features`
+#' @inheritParams corpus_sample
```
Comment (kbenoit) [inline R/dfm_sample.R:10]: The addition of `by` is welcome but works very differently for `margin = "features"`, so we might need to think this through.

**Review by kbenoit — CHANGES_REQUESTED** [review]
Looks good but we need to think through how the sampling works with the vectors and with `by` for feature dimensions of the dfm.

File: R/dfm_sample.R (hunk 1 above)
Comment (koheiw) [reply R/dfm_sample.R:10]: `by` has no effect for feature. I also think we don't need sampling for `features` as it is really easy: `dfmt[,sample(nfeat(dfmt), 1000)]`

Comment (kbenoit) [reply R/dfm_sample.R:10]: OK, I think we could simplify this then and make it more consistent by defining the `*_sample()` as operating on documents only. Let's deprecate margin with a warning and remove it from the next release. I suspect it gets very little use.

**Comment by kbenoit** [comment]
We should also make `tokens_sample()` consistent with the other two sample functions.

File: R/dfm_sample.R (hunk 1 above)
Comment (koheiw) [reply R/dfm_sample.R:10]: 👍

**Review by kbenoit — APPROVED** [review]

**Pull request merged**